server/asset/eth/common.go

```diff
@@ -57,32 +76,49 @@ func (ss SwapState) String() string {
 	return "unknown"
 }
 
-// DecodeCoinID decodes the coin ID into flags, a contract address, and secret hash.
-func DecodeCoinID(coinID []byte) (uint16, common.Address, []byte, error) {
+// DecodeCoinID decodes the coin ID into flags, a contract address which may be
+// zeroed, and hash that represents either a secret or txid depending on flags.
```

Comment:
If the meaning of "contract address which may be zeroed" is that a `CoinIDFlag` of `TxIDFlag` will return a pointer to a a zero-value `common.Address`, I think it would be most helpful to have that spelled out in the `DecodeCoinID` docs instead of just the `TxIDFlag` docs.

Reply:
Tried to be more informative.